### PR TITLE
ENH: change in CallbackPrintIteration.

### LINKED ISSUE #870: Let CallbackPrintIteration not printing every iteration
When I am running many iterations to solve a problem, I want to have iteration print in the terminal in order to see that the things are running. However, I might not need to see all of them. I suggest that we implement a `display_step` like in `CallbackShow`. What do you think?

### PR DESCRIPTION
**Pull request by aringh**
The user can specify how often the iteration number should be printed.

Edit: this fixes #870

### CODE REVIEW

File: odl/solvers/util/callback.py [hunk 1]
```diff
@@ -236,20 +236,25 @@ class CallbackPrintIteration(SolverCallback):
...
-    def __init__(self, text=None):
+    def __init__(self, text=None, display_step=1):
         """Initialize a new instance.
 
         Parameters
         ----------
         text : string
             Text to display before the iteration count. Default: 'iter ='
+        display_step : positive int, optional
+            Number of iterations between plots. Default: 1
```
Comment (kohr-h) [inline odl/solvers/util/callback.py]: "plots" doesn't make sense here. Use "output" or "print" or something like that.

**Comment by aringh** [comment]
Fixed :-) I am not sure I can merge, or maybe after two approved reviews?

**Comment by kohr-h** [comment]
You should be able to, just wait until CI has finished its job :-)

**Comment by aringh** [comment]
> You're not authorized to merge this pull request.

:p

**Comment by kohr-h** [comment]
Too bad :-)

**Pull request merged**